### PR TITLE
setup check to verify if the models are downloaded

### PR DESCRIPTION
**Pull request by kyteinsky**
For the case (#12), this issue does not solve it but provides aid to the admin by informing him/her about the missing binary executables and the models.

### CODE REVIEW

File: lib/SetupChecks/ModelFilesCheck.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		private IL10N $l10n,
+	) {
+	}
+
+	public function getCategory(): string {
+		return 'whisper';
+	}
+
+	public function getName(): string {
+		return 'Whisper Speech-To-Text';
```
Comment (marcelklehr) [inline lib/SetupChecks/ModelFilesCheck.php:25]: In the original PR in server getName() returns a text like `Checking for XXX of YYY app`

File: lib/SetupChecks/ModelFilesCheck.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+use OCP\SetupCheck\SetupResult;
+
+class ModelFilesCheck implements ISetupCheck {
+	public function __construct(
+		private IL10N $l10n,
+	) {
+	}
+
+	public function getCategory(): string {
+		return 'whisper';
```
Comment (marcelklehr) [inline lib/SetupChecks/ModelFilesCheck.php:21]: In the translate app I used 'ai' as category, shall we use that here too?

File: lib/SetupChecks/ModelFilesCheck.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+	public function run(): SetupResult {
+		$modelPathFor = fn (string $model) => __DIR__ . '/../../models/' . $model;
+		$modelPaths = array_map($modelPathFor, DownloadModelsService::AVAILABLE_MODELS);
+		$modelsExist = array_map('file_exists', $modelPaths);
+
+		if (in_array(true, $modelsExist, true)) {
+			return SetupResult::success($this->l10n->t('Whisper Speech-To-Text models are downloaded'));
+		}
+		return SetupResult::error($this->l10n->t('Whisper Speech-To-Text models are not downloaded, please run `occ stt_whisper:download-models [small%smedium%slarge]` to download them', ['|', '|']));
```
Comment (marcelklehr) [inline lib/SetupChecks/ModelFilesCheck.php:36]: Any reason for using params here? Also, `%s` is the wrong placeholder according to https://docs.nextcloud.com/server/latest/developer_manual/basics/front-end/l10n.html#translation

**Review by marcelklehr — CHANGES_REQUESTED** [review]

File: lib/SetupChecks/ModelFilesCheck.php (hunk 3 above)
Comment (kyteinsky) [reply lib/SetupChecks/ModelFilesCheck.php:36]: pipe is not supported in translation strings

**Comment by marcelklehr** [comment]
Might be worth it to also add checks for these while we're at it: https://github.com/nextcloud/stt_whisper/blob/main/src/components/ViewAdmin.vue#L18-L26

**Comment by kyteinsky** [comment]
yay, the transcribe test passes! Thanks Marcel.

File: lib/SetupChecks/BinaryFilesCheck.php [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		return 'Checking if binary files are present for Whisper Speech-To-Text app';
+	}
+
+	public function run(): SetupResult {
+		if ($this->isBinaryPresent($this->stt->getFfmpegBinary() . ' -version') === false) {
+			return SetupResult::error($this->l10n->t('Could not find the FFmpeg executable. You may need to set the path to a working executable manually in the "Whisper Speech-To-Text app" section of the admin settings.'));
+		}
+
+		if ($this->isBinaryPresent($this->stt->getWhisperBinary() . ' -h') === false) {
+			return SetupResult::error($this->l10n->t('Could not find the Whisper executable. You may need to compile Whisper yourself for it to run on this server\'s processor architecture.'));
```
Comment (marcelklehr) [inline lib/SetupChecks/BinaryFilesCheck.php:35]: can we rephrase 'Could not find'? It might be there but not working correctly

**Review by marcelklehr — APPROVED** [review]

**Comment by nickvergessen** [comment]
On our instance the setupchecks of this app print stuff on `occ setupcheck` which is not okay :(


```
root@cloud1:~# sudo -u www-data /var/www/…/occ setupcheck

usage: /var/www/…/apps/stt_whisper/lib/Service/../../bin/main [options] file0.wav file1.wav ...

options:
  -h,        --help              [default] show this help message and exit
  -t N,      --threads N         [4      ] number of threads to use during computation
  -p N,      --processors N      [1      ] number of processors to use during computation
  -ot N,     --offset-t N        [0      ] time offset in milliseconds
  -on N,     --offset-n N        [0      ] segment index offset
  -d  N,     --duration N        [0      ] duration of audio to process in milliseconds
  -mc N,     --max-context N     [-1     ] maximum number of text context tokens to store
  -ml N,     --max-len N         [0      ] maximum segment length in characters
  -sow,      --split-on-word     [false  ] split on word rather than on token
  -bo N,     --best-of N         [5      ] number of best candidates to keep
  -bs N,     --beam-size N       [5      ] beam size for beam search
  -wt N,     --word-thold N      [0.01   ] word timestamp probability threshold
  -et N,     --entropy-thold N   [2.40   ] entropy threshold for decoder fail
  -lpt N,    --logprob-thold N   [-1.00  ] log probability threshold for decoder fail
  -debug,    --debug-mode        [false  ] enable debug mode (eg. dump log_mel)
  -tr,       --translate         [false  ] translate from source language to english
  -di,       --diarize           [false  ] stereo audio diarization
  -tdrz,     --tinydiarize       [false  ] enable tinydiarize (requires a tdrz model)
  -nf,       --no-fallback       [false  ] do not use temperature fallback while decoding
  -otxt,     --output-txt        [false  ] output result in a text file
  -ovtt,     --output-vtt        [false  ] output result in a vtt file
  -osrt,     --output-srt        [false  ] output result in a srt file
  -olrc,     --output-lrc        [false  ] output result in a lrc file
  -owts,     --output-words      [false  ] output script for generating karaoke video
  -fp,       --font-path         [/System/Library/Fonts/Supplemental/Courier New Bold.ttf] path to a monospace font for karaoke video
  -ocsv,     --output-csv        [false  ] output result in a CSV file
  -oj,       --output-json       [false  ] output result in a JSON file
  -ojf,      --output-json-full  [false  ] include more information in the JSON file
  -of FNAME, --output-file FNAME [       ] output file path (without file extension)
  -np,       --no-prints         [false  ] do not print anything other than the results
  -ps,       --print-special     [false  ] print special tokens
  -pc,       --print-colors      [false  ] print colors
  -pp,       --print-progress    [false  ] print progress
  -nt,       --no-timestamps     [false  ] do not print timestamps
  -l LANG,   --language LANG     [en     ] spoken language ('auto' for auto-detect)
  -dl,       --detect-language   [false  ] exit after automatically detecting language
             --prompt PROMPT     [       ] initial prompt
  -m FNAME,  --model FNAME       [models/ggml-base.en.bin] model path
  -f FNAME,  --file FNAME        [       ] input WAV file path
  -oved D,   --ov-e-device DNAME [CPU    ] the OpenVINO device used for encode inference
  -ls,       --log-score         [false  ] log best decoder scores of tokens
  -ng,       --no-gpu            [false  ] disable GPU

	dav:
		✓ DAV system address book: No outstanding DAV system address book sync.
	system:
		ℹ Brute-force Throttle: Your remote address could not be determined.
		✓ File locking
		✓ Memcache: Configured
		✓ PHP modules
		✓ Architecture: 64-bit
	notifications:
		✓ Push notifications - Fair use policy
	security:
		✓ Old user imported certificates
		ℹ Forwarded for headers: Your remote address could not be determined.
		✓ Old server-side-encryption: Disabled
		✓ PHP version: You are currently running PHP 8.1.2-1ubuntu2.14.
		✓ Random generator: Secure
	database:
		✓ Database missing columns: None
		⚠ Database missing indices: The database is missing some indexes. Due to the fact that adding indexes on big tables could take some time they were not added automatically. By running "occ db:add-missing-indices" those missing indexes could be added manually while the instance keeps running. Once the indexes are added queries to those tables are usually much faster.
Missing optional index "mail_messages_msgid_idx" in table "mail_messages".
		✓ Database missing primary keys: None
		✓ Database version: 10.6.12-MariaDB-0ubuntu0.22.04.1
		✓ Database transaction isolation level: Read committed
	config:
		ℹ Default phone region: Your installation has no default phone region set. This is required to validate phone numbers in the profile settings without a country code. To allow numbers without a country code, please add "default_phone_region" with the respective ISO 3166-1 code of the region to your config file.
		✓ Email test: Email test was successfully sent
		✓ Configuration file access rights: Nextcloud configuration file is writable
	network:
		✓ Internet connectivity
	php:
		✓ PHP default charset: UTF-8
		✓ Freetype: Supported
		✓ PHP getenv
		✓ PHP memory limit: 512 MB
		✓ PHP "output_buffering" option: Disabled
	ai:
		✓ Checking if models are downloaded for Whisper Speech-To-Text app: Whisper Speech-To-Text models are downloaded
		✓ Checking if binary files are present for Whisper Speech-To-Text app: Whisper Speech-To-Text app binary files are present
		✓ Checking whether your server supports AVX CPU instructions.: Your server supports AVX CPU instructions, which are needed by the translate app.
		✓ Checking whether translate app models are downloaded: Translate app models are downloaded and available for use.
		✓ Checking whether the Node.js executable of the translate app works: The node.js executable for the translate app is working.
	ldap:
		✓ Invalid LDAP UUIDs: None found
```

**Pull request merged**